hibernate-core/src/main/java/org/hibernate/boot/registry/classloading/internal/ClassLoaderServiceImpl.java

```diff
@@ -113,7 +114,12 @@ public static ClassLoaderServiceImpl fromConfigSettings(Map configValues) {
 		addIfSet( providedClassLoaders, AvailableSettings.HIBERNATE_CLASSLOADER, configValues );
 		addIfSet( providedClassLoaders, AvailableSettings.ENVIRONMENT_CLASSLOADER, configValues );
 
-		return new ClassLoaderServiceImpl( providedClassLoaders,TcclLookupPrecedence.AFTER );
+		TcclLookupPrecedence providedPrecedence = TcclLookupPrecedence.AFTER;
```

Comment:
```
final TcclLookupPrecedence tcclPrecedence = TcclLookupPrecedence.fromSetting( configValues );
return new ClassLoaderServiceImpl( providedClassLoaders,providedPrecedence );
```

```
public enum TcclLookupPrecedence {
    ...

    public static TcclLookupPrecedence from(Map<?,?> settings) {
        final String explicitSetting = (String) settings.get( AvailableSettings.TC_CLASSLOADER );
        if ( explicitSetting == null ) {
            return AFTER;
        }

        if ( NEVER.name().equalsIgnoreCase( explicitSetting ) ) {
            return NEVER;
        }
		
        if ( BEFORE.name().equalsIgnoreCase( explicitSetting ) ) {
            return BEFORE;
        }

        return AFTER;
    }
}
```

Reply:
I am glad to help!

1.  About static resolution. I copied code from this line [here](https://github.com/hibernate/hibernate-orm/blob/67fd24ecff9f9a9a2e848805c04590b4b4ee2b3e/hibernate-core/src/main/java/org/hibernate/jpa/boot/internal/EntityManagerFactoryBuilderImpl.java#L447). If I add `TcclLookupPrecedence.from(Map)` method, should I also change and that line 447?

2. About test - unfortunately I know about hibernate too little and I just can't make a test. I need someone's help.

Reply:
> 1. About static resolution. I copied code from this line [here](https://github.com/hibernate/hibernate-orm/blob/67fd24ecff9f9a9a2e848805c04590b4b4ee2b3e/hibernate-core/src/main/java/org/hibernate/jpa/boot/internal/EntityManagerFactoryBuilderImpl.java#L447). If I add `TcclLookupPrecedence.from(Map)` method, should I also change and that line 447?

Awesome, thanks!  Looks good.  With regard to line 447, in my opinion that should change also.

> 2. About test - unfortunately I know about hibernate too little and I just can't make a test. I need someone's help.

We have templates you can use to create tests.  They already do most of the heavy lifting for you.  See https://github.com/hibernate/hibernate-test-case-templates

We won't apply this without a test.  But I'll try to find some time to create one.  Would be awesome if you could write the test though.

Reply:
I would be very thankful if you provide a test. If I make a pull request again I will study it.

Reply:
@sebersole What about line 445 - 447, maybe in TcclLookupPrecedence#from we should check if map is Properties and if true then to use getProperty method? [Here](https://stackoverflow.com/a/11104420/5057736) it is said it is said that get() not always returns the same as getProperty(). It depends on how the properties were created. What do you think?

Reply:
@sebersole Any ideas?